src/utils.ts

```diff
@@ -24,3 +25,62 @@ export const toPercent = (value: number, min: number, max: number): number => {
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const isString = (value: any): value is string => typeof value === 'string'
+
+export const compareStrings = (
+  referenceStr: string,
+  compareString: string,
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  locales?: string | string[],
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  options?: Intl.CollatorOptions | undefined
+): number => referenceStr.localeCompare(compareString, locales, options)
+
+const ascendingSortByGroupAndValues = (
+  option1: Readonly<Option>,
+  option2: Readonly<Option>
+): number => {
+  const definedOption1 = option1.group ?? ''
+  const definedOption2 = option2.group ?? ''
+
+  if (definedOption1 > definedOption2) {
+    return 1
+  }
+
+  if (definedOption1 < definedOption2) {
+    return -1
+  }
+
+  return compareStrings(option1.value, option2.value)
+}
+
+const descendingSortByGroupAndValues = (option1: Option, option2: Option): number => {
+  const definedOption1 = option1.group ?? ''
+  const definedOption2 = option2.group ?? ''
+  if (definedOption1 < definedOption2) {
+    return 1
+  }
+  if (definedOption1 > definedOption2) {
+    return -1
+  }
+  return option2.value.localeCompare(option1.value)
```

Comment:
Why using a `compareStrings` for ascending, and not the same function for descending?